Sources/TracingMacros/TracedMacro.swift

```diff
@@ -0,0 +1,25 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Distributed Tracing open source project
+//
+// Copyright (c) 2020-2024 Apple Inc. and the Swift Distributed Tracing project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Distributed Tracing project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+@_exported import ServiceContextModule
+import Tracing
+
+#if compiler(>=6.0)
+@attached(body)
+public macro Traced(
+    _ operationName: String? = nil,
+    context: ServiceContext? = nil,
```

Comment:
Hmm, technically is the default `.current` rather than `nil`?

Should the type be the `@autoclosure () -> ServiceContext` as well, so we avoid reading the local when tracing is disabled (not bootstrapped)

Reply:
I'm not entirely sure about the answer here since this is my first major macro.
- For the default: I'm using `nil` just to delegate the default handling to `withSpan` itself—we don't even construct the syntax to pass a parameter when it's omitted here. I don't fully understand how default parameters are even passed to the macro.
- For the type: since this is a macro it's handled via syntax expansion, so the type is used for checking the interface, not as a runtime type. When the expression is interpolated into the call, I think it will end up becoming an autoclosure at the `withSpan` callsite?

Reply:
I did some experimentation: The default value isn't passed to the macro at all, it seems to just be for documentation purposes?

Reply:
Since it seems to be only documentation—do we want to duplicate the default values from `withSpan` here, or cross-reference to say "look at the default values in the function documentation"? I can see arguments for both. I leaned towards omitting them here to not duplicate code across modules, but it's already duplicated across many overload so it's probably worth it for the documentation's sake.

Reply:
Oh good point about the autoclosure not mattering! 

the nil maybe is fine after all, since it just means "don't pass anything" makes sense to me

For the docs: i think it's fine to not duplicate the docs, just a short note that it is passed to the `withSpan` will be enough

Reply:
I see, yeah let's document to just look at the real method maybe